sql/2019/06_Fonts/06_09.sql

```diff
@@ -0,0 +1,32 @@
+#standardSQL
+
+-- counts the number of pages given number of font-face
+
+WITH
+  response_bodies AS (
+    SELECT * FROM `httparchive.response_bodies.2019_07_01_desktop`
+    UNION ALL
+    SELECT * FROM `httparchive.response_bodies.2019_07_01_mobile`
+  ),
+  temp AS (
+    SELECT ARRAY_LENGTH(REGEXP_EXTRACT_ALL(body, r"\@font-face\s*\{[^}]+\}")) AS font_display_arr, url, page
```

Comment:
This metric description is asking about two different things: fonts resources and font faces. Let's put resources in 06_09a.sql and faces in 06_09b.sql so we can have them separated as needed. The font faces problem is probably best answered with a CSS-specific solution that we're still working out for the CSS chapter. Related: https://discuss.httparchive.org/t/analyzing-stylesheets-with-a-js-based-parser/1683

Reply:
Chrome also has counters for every time it encounters a specific web API. We expose these in HTTP Archive as blinkFeatures.

There is a counter called [CSSAtRuleFontFace](https://www.chromestatus.com/metrics/feature/timeline/popularity/970) that could be helpful as an inexpensive way of detecting when those are used. Other relevant feature counters: [DocumentFonts](https://www.chromestatus.com/metrics/feature/timeline/popularity/440) (not sure what this one is detecting) and [FontFaceConstructor](https://www.chromestatus.com/metrics/feature/timeline/popularity/989) (JS API)